ds-collector/README.md

```diff
@@ -1,70 +1,242 @@
+```
+./ds-collector -h
 ```
+The help displayed comes from the `ds-collector` scripts [`hjalp`](https://github.com/datastax/diagnostic-collection/blob/master/ds-collector/ds-collector#L727-L752) function.
 
 Directions
 ----------
 
 * Extract the three files to your central host (e.g. bastion or jumpbox). Do not run this script on a Cassandra/DSE node.
-* Copy the generated encryption key into the same directory as the ds-collector script.
+```
+tar -xvf ds-collector.*.tar.gz
+cd collector
+```
+
+* If an encryption key has been provided, copy it to the same extracted directory (where the ds-collector script is found).
+```
+cp <path-to-encryption-file>/*_secret.key .
+```
+
 * Make sure the correct permissions are set on the ds-collector so it is executable.
-    ```
-    chmod +x ds-collector
-    ```
-* Update `collector.conf` variable "userName" to the name of a user who can log into the nodes.
-* You must be able to ssh from the central host to each node, using ssh agent forwarding, an identity file, or sshpass.
-* Update all other options in `collector.conf` accordingly.
-* Test the connection with command
-    ```
-    ./ds-collector -T -f collector.conf -n <CASSANDRA_CONTACT_POINT>
-    ```
-* If test returns `NOTOK` please notify DataStax via email including the message.
-* Run the script to collect data and script will automatically upload to a secure S3 bucket.
-    ```
-    ./ds-collector -X -f collector.conf -n <CASSANDRA_CONTACT_POINT>
-    ```
+```
+ls -l ds-collector
+chmod +x ds-collector
+```
+
+* Update `collector.conf`, read through the file and configure each option if and as neccessary. Do not change `issueId`, `keyId`, or `keySecret`.
+
+* Test the connection with command. `-T` invokes test mode. `-f …` is used to specify the configuration file. `-n …` is used to provide one alive Cassandra/DSE node in the cluster to contact with to discover the other nodes in the cluster. The `-p` option can also be added to first test with a ping to the contact node specified.
+```
+./ds-collector -T -f collector.conf -n <CASSANDRA_CONTACT_NODE>
+```
 
+* If test returns `NOTOK` please notify us including the message.
+
+* Run the script to collect data. `-X` invokes the execute mode. Diagnostic snapshots will be collected from all reachable nodes. Also, if enabled the script will encrypt the diagnostic tarballs, and upload them to a secure S3 bucket. 
+```
+./ds-collector -X -f collector.conf -n <CASSANDRA_CONTACT_NODE>
+```
+
+* If the run returns `NOTOK` please notify us including the message.
+
+
+The script, with the same configuration and encryption key, can be run multiple times as diagnostic snapshots are timestamped.
+
+Recipes
+=======
+
+* To run the collector against a single node only, use the `-d` option.
+```
+./ds-collector -X -d -f collector.conf -n <CASSANDRA_NODE>
+```
+
+* To run the collector against a specified list of nodes only, disabling the discovery of all nodes, use the `collector.hosts` file.
```

Comment:
Do we have any notes on the format of this file? that catches people out, \n separated? comma separated? etc

Reply:
fixed. (it was already inside the file too)